pkg/resource/deploy/step_generator.go

```diff
@@ -1157,8 +1157,8 @@ func (sg *stepGenerator) generateStepsFromDiff(
 	}
 
 	// If there were changes check for a replacement vs. an in-place update.
-	if diff.Changes == plugin.DiffSome {
-		if diff.Replace() {
+	if diff.Changes == plugin.DiffSome || old.PendingReplacement {
+		if diff.Replace() || old.PendingReplacement {
```

Comment:
I don't think this check is needed. We would have hit this check in the first update, and if the protect bit has been added since then it doesn't matter because we've already deleted this resource, if its PendingReplacement we're just going to remove it from state, there isn't anything left to delete.

Reply:
Thinking on this more, I think this check _is_ needed incase the goal state has changed such that it wouldn't be a replace _now_ but it was a replace when we did the delete.